stock_request_analytic/tests/test_stock_request_analytic.py

```diff
@@ -0,0 +1,142 @@
+                        "analytic_account_id": analytic.id,
+                        "company_id": company.id,
+                        "warehouse_id": self.warehouse.id,
+                        "location_id": self.demand_loc.id,
+                        "expected_date": expected_date,
+                    },
+                )
+            ],
+        }
+        return vals
+
+    def prepare_order_request_multi_analytic(self, analytic1, analytic2, company):
+        expected_date = fields.Datetime.now()
+        vals = {
+            "company_id": company.id,
+            "warehouse_id": self.warehouse.id,
+            "location_id": self.demand_loc.id,
+            "expected_date": expected_date,
+            "stock_request_ids": [
+                (
+                    0,
+                    0,
+                    {
+                        "product_id": self.pizza.id,
+                        "product_uom_id": self.pizza.uom_id.id,
+                        "product_uom_qty": 5.0,
+                        "analytic_account_id": analytic1.id,
+                        "company_id": company.id,
+                        "warehouse_id": self.warehouse.id,
+                        "location_id": self.demand_loc.id,
+                        "expected_date": expected_date,
+                    },
+                ),
+                (
+                    0,
+                    0,
+                    {
+                        "product_id": self.pizza.id,
+                        "product_uom_id": self.pizza.uom_id.id,
+                        "product_uom_qty": 5.0,
+                        "analytic_account_id": analytic2.id,
+                        "company_id": company.id,
+                        "warehouse_id": self.warehouse.id,
+                        "location_id": self.demand_loc.id,
+                        "expected_date": expected_date,
+                    },
+                ),
+            ],
+        }
+        return vals
+
+    def test_stock_analytic(self):
+        vals = self.prepare_order_request_analytic(self.analytic, self.main_company)
+        order = self.env["stock.request.order"].create(vals)
+        req = order.stock_request_ids
+        order.action_confirm()
+        self.assertEqual(req.move_ids.mapped("analytic_account_id"), self.analytic)
+        self.assertEqual(order.analytic_count, 1)
+        action = order.action_view_analytic()
+        self.assertTrue(action["res_id"], self.analytic.id)
+        action2 = self.analytic.action_view_stock_request()
+        self.assertTrue(action2["res_id"], order.id)
+
+    def test_stock_multi_analytic(self):
+        vals = self.prepare_order_request_multi_analytic(
+            self.analytic, self.analytic3, self.main_company
+        )
+        order = self.env["stock.request.order"].create(vals)
+        order.action_confirm()
+        self.assertEqual(order.analytic_count, 2)
+
+    def test_company(self):
+        with self.assertRaises(ValidationError):
```

Comment:
Tests need to be adapted if the suggestion above is included
```suggestion
        with self.assertRaises(UserError):
```